.github/workflows/build_and_deploy_dev.yml

```diff
@@ -0,0 +1,86 @@
+name: Build and Deploy to dev service app
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [elr/gh-action-dev-deploy]
+    # branches: [master]
+
+env:
+  CODE_ENVIRONMENT: dev
+  APPLICATION_TYPE: api
+  AZURE_WEBAPP_NAME: clearlydefined-api-dev
+  DOCKER_IMAGE_NAME: ghcr.io/${{ github.repository_owner }}/${{ github.repository }}-dev
+  # SEMVER: '' # if you want to use semver, you will need to add a tag to the Docker build
+
+jobs:
+  build-and-deploy:
+    name: Build and Deploy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Log into ghcr registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }} # user that kicked off the action
+          password: ${{ secrets.GITHUB_TOKEN }} # token created when the action launched (short lived)
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          file: Dockerfile
+          tags: |
+            ${{ env.DOCKER_IMAGE_NAME }}:latest 
+            ${{ env.DOCKER_IMAGE_NAME }}:${{ github.sha }}
+          labels: |
+            env=${{ env.CODE_ENVIRONMENT }}
+            type=${{ env.APPLICATION_TYPE }}
+
+      - name: Login for Azure cli commands
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Set DOCKER configs in Azure web app
+        uses: Azure/appservice-settings@v1
+        with:
+          app-name: ${{ env.AZURE_WEBAPP_NAME }}
+          app-settings-json: |
+            [
+              {
+                    "name": "DOCKER_CUSTOM_IMAGE_NAME",
+                    "value": "${{ env.DOCKER_IMAGE_NAME }}:latest }}",
```

Comment:
I think it is problematic to use the 'latest' tag for deployment. If we'll have to debug the deployment, it will not be immediately clear which exactly tag was 'latest' during the deployment time. I think it's better to use a fixed image version here, which you luckily have access to, because you use git sha sum for it.

Reply:
Thanks for the feedback.  What you are saying makes sense.  I will remove `:latest` tag and only use the sha for development.  For production, the plan was already to only use the semver version.